### PR TITLE
Refactor Store

### PR DESCRIPTION
**Pull request by ryanhobsonsmith**
Refactor state flow so that we don't rely on URL hash as a source of truth and instead let the vue store update the URL hash whenever it's state changes.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
-  "presets": ["env","es2015","stage-0"]
+  "presets": ["env","es2015","stage-0"],
+  "sourceMaps": "inline"
```
Comment (stuartsan) [inline .babelrc]: I'm assuming this won't make sourcemaps inlined in prod?

File: src/components/Header.js [hunk 2]
```diff
@@ -26,6 +24,9 @@ export default function(store){
       filters() {
         return store.state.filters;
       },
+      zipcode() {
```
Comment (stuartsan) [inline src/components/Header.js]: Just realized that whatever is referencing zipcode could also just get it out of `filters.zipcode`, but either way

File: src/components/Header.js [hunk 3]
```diff
@@ -44,14 +45,12 @@ export default function(store){
...
         this.isFilterEventsOpen = !this.isFilterEventsOpen;
       },
-      search() {
-        setHash({ zipcode: this.zipcode });
-      }
-    },
-    watch: {
-      zipcode(newZipcode) {
+      search(e) {
+        let newZipcode = e.target.value;
```
Comment (stuartsan) [inline src/components/Header.js]: Nit: could use `const` (also console statements :) )

File: src/store.js [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
 import Vuex from 'vuex';
 import xhr from 'xhr';
-import { getHash, onHashChange } from 'src/util/url-hash';
+import { computeFilteredEvents } from 'src/util/events';
+import querystring from 'querystring';
 
-Vue.use(Vuex);
+//Save the initial hash of the window when Vue initializes.
+//We'll use these values to populate the initial store filter values
+let initialHash = querystring.parse(window.location.hash.replace(/^#/, ''))
```
Comment (stuartsan) [inline src/store.js]: same re: `const`

File: src/store.js [hunk 5]
```diff
@@ -1,16 +1,30 @@
...
+
+//A small Vuex plugin that will update the browsers location hash whenever
+//the setFilters mutation occurs
+const hashUpdaterPlugin = (store) => {
```
Comment (stuartsan) [inline src/store.js]: This is sweet

File: src/store.js [hunk 6]
```diff
@@ -41,16 +55,22 @@ const store = new Vuex.Store({
...
-      state.filters = filters;
-    },
     viewToggled(state) {
       state.view = state.view === 'map' ? 'list' : 'map';
+    },
+    setFilters(state, filters) {
+      state.filters = {...state.filters, ...filters}
+    },
+  },
+  getters: {
```
Comment (stuartsan) [inline src/store.js]: I wonder if store getters are memoized in the same way that computed props are in components....that would be cool.

File: src/store.js [hunk 7]
```diff
@@ -41,16 +55,22 @@ const store = new Vuex.Store({
...
     }
-  }
+  },
+  plugins: [hashUpdaterPlugin]
 });
 
 // When url filters change, update the store.
-onHashChange((filters) => store.commit('filtersReceived', filters));
-
+// onHashChange((filters) => store.commit('filtersReceived', filters));
```
Comment (stuartsan) [inline src/store.js]: These two safe to delete?

File: src/store.js (hunk 6 above)
Comment (ryanhobsonsmith) [reply src/store.js]: Hmm, yeah I always assumed they were, but now that I've re-read docs I don't see it called out explicitly so I could be mistaken.

File: src/store.js (hunk 7 above)
Comment (ryanhobsonsmith) [reply src/store.js]: Yup, we should get rid of those

File: .babelrc (hunk 1 above)
Comment (ryanhobsonsmith) [reply .babelrc]: Oh, yeah good point. I don't think this really helped actually. I'll just remove it

**Review by stuartsan — APPROVED** [review]

**Pull request merged**